src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -250,6 +250,15 @@ void SubscribeVehicleDataRequest::on_event(const event_engine::Event& event) {
       result_code = mobile_apis::Result::IGNORED;
       response_info = "Already subscribed on some provided VehicleData.";
     }
+
+    if (!vi_waiting_for_subscribe_.empty()) {
+      LOG4CXX_DEBUG(logger_, "Subscribing to all pending VehicleData");
+      VehicleInfoSubscriptions::const_iterator key =
```

Comment:
@JackLivio there is no reasons to place locks around this variable.
`vi_waiting_for_subscribe_.insert` is in `SubscribeVehicleDataRequest::Run` which checks all request param and then sends request to HMI, and the second occurrence is in `SubscribeVehicleDataRequest::on_event` which is called when response from HMI will be received so its physically not possible to access this variable concurrently because response could not be received until request will not be sent.